src/User.js

```diff
@@ -13,6 +13,24 @@ class User {
         let today = moment(Date.now());
         return this.trips.filter(trip => moment(trip.date).isBefore(today) && moment(trip.date).isAfter(today.subtract(1, 'year')))
     }
+
+    searchApprovedTrips(timeFrame) {
+        let today = moment(Date.now());
+        let searchedTrips;
+        if (timeFrame === 'current') {
+            searchedTrips = this.trips.filter(trip => moment(trip.date).add(trip.duration, 'day').isAfter(today) && moment(trip.date).add(trip.duration, 'day').isBefore(today))
+
+        } else if (timeFrame === 'upcoming') {
+            searchedTrips =  this.trips.filter(trip => moment(trip.date).add(trip.duration, 'day').isAfter(today))
+        } else {
+            searchedTrips = this.trips.filter(trip => moment(trip.date).add(trip.duration, 'day').isBefore(today))
+        }
+        return searchedTrips
+    }
```

Comment:
I'm not sure if I'm reading this correctly, but one thing I noticed in the data is that there are trips that are marked pending but also have a start date in the past. If you only want to return past trips the user has actually taken, you'll want to filter for trip.status as well. I decided to return all pending trips thinking that in a future iteration that would allow a user to pursue a refund from the travel agent. *Wait* I just saw searchApprovedTrips, so maybe you already have filtered out all pending trips in a previous method? In which case, nevermind, lol.